src/server/frontend_wayland/window_wl_surface_role.cpp

```diff
@@ -40,6 +40,27 @@ namespace ms = mir::scene;
 namespace msh = mir::shell;
 namespace geom = mir::geometry;
 
+namespace
+{
+geom::Size const max_possible_size{
+    std::numeric_limits<int>::max(),
+    std::numeric_limits<int>::max()};
+
+/// Clears pending if it holds a value different than cache
+/// sets cache to pending and leaves pending alone if it holds a different value
+template<typename T>
+inline void clear_pending_if_unchanged(mir::optional_value<T>* pending, T* cache)
```

Comment:
In C++ I'd use references, not pointers.

Reply:
I wanted to make it obvious from the call site that the function modifies the parameters

Reply:
That's a C convention, not C++. You don't write, for example, `std::swap(&left, &right)`

Reply:
Alright. I don't like it, but won't argue with language conventions